File: 알고리즘/CodingTest/src/Programmers/level2/JadenCase.java
package Programmers.level2;

// https://programmers.co.kr/learn/courses/30/lessons/12951

public class JadenCase {
    public String solution(String s) {
        String answer = "";
        String[] sp = s.toLowerCase().split("");

        boolean flag = true; // 첫 글자는 항상 대문자

        for(String str : sp) {
            answer += flag ? str.toUpperCase() : str; // flag == true 이면 대문자로
            flag = str.equals(" ")? true : false; // flag가 공백일 경우 flag == true로 변경
        }
        return answer;
    }
}

// 코드 참고: https://hj-bank.tistory.com/entry/%ED%94%84%EB%A1%9C%EA%B7%B8%EB%9E%98%EB%A8%B8%EC%8A%A4-JadenCase-JAVA

File: Algorithm/Algorithm_py/programmers/programmers2/JadenCase.py
# https://programmers.co.kr/learn/courses/30/lessons/12951?language=python3
def solution(s):
  answer = ''
  x = 0
  
  for idx, value in enumerate(s):
    if idx == 0:
      if value.isalpha():
        answer += value.upper()
      else:
        answer += value
    else:
      if value == ' ':
        x = 1
        answer += value
      elif value != ' ' and x == 1:
        answer += value.upper()
        x = 0
      else:
        answer += value.lower()

  return answer